apps/backend/src/collectivites/discussions/infrastructure/discussion-message.repository.impl.ts

```diff
@@ -0,0 +1,140 @@
+    discussionMessage: CreateDiscussionMessageType,
+    tx?: Transaction
+  ): Promise<Result<DiscussionMessage>> {
+    try {
+      const result = await (tx ?? this.databaseService.db)
+        .insert(discussionMessageTable)
+        .values(discussionMessage)
+        .returning();
+      if (!result || result.length === 0) {
+        this.logger.error(
+          `Error creating discussion message: ${JSON.stringify(
+            discussionMessage
+          )}`
+        );
+        return {
+          success: false,
+          error: DiscussionErrorEnum.DATABASE_ERROR,
+        };
+      }
+      const [createdDiscussionMessage] = result;
+      this.logger.log(
+        `Successfully created discussion message ${createdDiscussionMessage.id}`
+      );
+      // Fetch the createdByNom field for the new message to match the structure of DiscussionMessage
+      const createdByNomResult = await (tx ?? this.databaseService.db)
+        .select({
+          createdByNom: sql<string | null>`
+            CASE
+              WHEN ${userTable.prenom} IS NULL AND ${userTable.nom} IS NULL
+                THEN NULL
+              ELSE TRIM(CONCAT(COALESCE(${userTable.prenom}, ''), ' ', COALESCE(${userTable.nom}, '')))
+            END
+          `,
+        })
+        .from(userTable)
+        .where(eq(userTable.userId, createdDiscussionMessage.createdBy!))
+        .limit(1);
+
+      const createdByNom =
+        createdByNomResult && createdByNomResult[0]
+          ? createdByNomResult[0].createdByNom
+          : null;
+
+      return {
+        success: true,
+        data: {
+          ...createdDiscussionMessage,
+          createdByNom,
+        },
+      };
+    } catch (error) {
+      this.logger.error(`Error creating discussion message: ${error}`);
+      return {
+        success: false,
+        error: DiscussionErrorEnum.DATABASE_ERROR,
+      };
+    }
+  }
+
+  async findByDiscussionIds(
+    discussionIds: number[]
+  ): Promise<Result<DiscussionMessage[]>> {
+    try {
+      const discussionMessages = await this.databaseService.db
+        .select({
+          ...getTableColumns(discussionMessageTable),
+          createdByNom: sql<
+            string | null
+          >`CASE WHEN ${userTable.prenom} IS NULL AND ${userTable.nom} IS NULL THEN NULL ELSE TRIM(CONCAT(COALESCE(${userTable.prenom}, ''), ' ', COALESCE(${userTable.nom}, ''))) END`,
+        })
+        .from(discussionMessageTable)
+        .leftJoin(
+          userTable,
+          eq(discussionMessageTable.createdBy, userTable.userId)
+        )
+        .where(inArray(discussionMessageTable.discussionId, discussionIds));
+      return {
+        success: true,
+        data: discussionMessages,
+      };
+    } catch (error) {
+      this.logger.error(
+        `Error finding discussion messages by discussion ids: ${error}`
+      );
+      return {
+        success: false,
+        error: DiscussionErrorEnum.DATABASE_ERROR,
+      };
+    }
+  }
+
```

Comment:
ici, on ne manipule pas uniquement l'objet discussion pure donc on est déjà en train de s'adapter à un besoin UI donc ça ne rentrerait pas complètement dans un repository mais dans un query service par exemple (cf cet [article](https://medium.com/@mr.karegar/handling-complex-queries-in-domain-driven-design-ddd-f6bbadffebe4))